### PR TITLE
Fix forcing dynamic batch in case if non-compatible plugins

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

**Comment by sovrasov** [comment]
This fix requires validation, because I don't have sufficient hardware and don't know exactly whether it will help or not.

File: demos/smart_classroom_demo/main.cpp [hunk 1]
```diff
@@ -419,6 +419,19 @@ std::map<int, int> GetMapFaceTrackIdToLabel(const std::vector<Track>& face_track
...
+        if (ie.GetConfig(device, CONFIG_KEY(DYN_BATCH_ENABLED)).as<std::string>() != PluginConfigParams::YES)
+            is_supported = false;
+    }
+    catch(const std::exception& error)  {
+        is_supported = false;
+    }
+    return is_supported;
+}
+
+
```
Comment (Wovchena) [inline demos/smart_classroom_demo/main.cpp]: Extra line, I guess.

**Comment by Wovchena** [comment]
It runs well for me for `CPU`, `GPU` or `MYRIAD` as `-d_reid` or `-d_lm`.

File: demos/smart_classroom_demo/main.cpp [hunk 2]
```diff
@@ -575,9 +587,12 @@ int main(int argc, char* argv[]) {
 
         // Load face reid
         CnnConfig reid_config(fr_model_path, fr_weights_path);
-        reid_config.max_batch_size = 16;
         reid_config.enabled = face_config.enabled && !fr_model_path.empty() && !lm_model_path.empty();
         reid_config.deviceName = FLAGS_d_reid;
+        if (!checkDynamicBatchSupport(ie, FLAGS_d_reid))
+            reid_config.max_batch_size = 1;
+        else
```
Comment (IRDonch) [inline demos/smart_classroom_demo/main.cpp]: A double negative here. You should flip the `if` and `else` branches.

File: demos/smart_classroom_demo/src/cnn.cpp [hunk 3]
```diff
@@ -68,7 +68,8 @@ void CnnDLSDKBase::InferBatch(
             matU8ToBlob<uint8_t>(frames[batch_i + b], input, b);
         }
 
-        infer_request_.SetBatch(current_batch_size);
+        if (config_.max_batch_size != 1)
+          infer_request_.SetBatch(current_batch_size);
```
Comment (IRDonch) [inline demos/smart_classroom_demo/src/cnn.cpp]: Wrong size indentation.

File: demos/smart_classroom_demo/main.cpp [hunk 4]
```diff
@@ -419,6 +419,18 @@ std::map<int, int> GetMapFaceTrackIdToLabel(const std::vector<Track>& face_track
     return face_track_id_to_label;
 }
 
+bool checkDynamicBatchSupport(const Core& ie, const std::string& device)  {
+    bool is_supported = true;
```
Comment (IRDonch) [inline demos/smart_classroom_demo/main.cpp]: Useless variable. You could just write `return true`/`false` where needed.

**Pull request merged**